File: R/LOOCV.R
source("HNMDRP.R")
LOOCV <- function(transCD, simCC, simDD, simTT, transDT, decay, min_sim){
	interaction_CD = as.matrix(transCD)
	simC = as.matrix(simCC)
	simD = as.matrix(simDD)
	simT = as.matrix(simTT)
	interaction_DT = as.matrix(transDT)
	
	interaction_CD_ori = interaction_CD
	score_ori = HNMDRP(interaction_CD, simC, simD, simT, interaction_DT, decay, min_sim)
	index = which(1==interaction_CD)
	for(i in 1:length(index)){
		interaction_CD[index[i]] = 0
    		score = HNMDRP(interaction_CD, simC, simD, simT, interaction_DT, decay, min_sim)
    		score_ori[index[i]] = score[index[i]]
    		interaction_CD = interaction_CD_ori
	}
	predict_score = as.numeric(matrix(score_ori))
	return (predict_score)
}

File: Matlab/LOOCV.m
function [predict_score] = LOOCV(transCD,simC,simD,simT,transDT,decay,min_sim)
A_ori = transCD;
[scoreCD_ori,scoreDT] = HNMDRP(transCD,simC,simD,simT,transDT,decay,min_sim);
index = find(1 == A_ori);
for i = 1:length(index)
    i
    transCD(index(i)) = 0;
    [scoreCD,scoreDT] = HNMDRP(transCD,simC,simD,simT,transDT,decay,min_sim);
    scoreCD_ori(index(i)) = scoreCD(index(i));
    scoreCD_ori(index(i));
    transCD = A_ori;
end
predict_score = scoreCD_ori;

